wallet/bdkgo/bdkgo.go

```diff
@@ -0,0 +1,1548 @@
+package bdkgo
+
+// #include <bdkgo.h>
+import "C"
+
+import (
+	"bytes"
+	"encoding/binary"
+	"fmt"
+	"io"
+	"math"
+	"runtime"
+	"sync/atomic"
+	"unsafe"
+)
+
+type RustBuffer = C.RustBuffer
+
+type RustBufferI interface {
+	AsReader() *bytes.Reader
+	Free()
+	ToGoBytes() []byte
+	Data() unsafe.Pointer
+	Len() int
+	Capacity() int
+}
+
+func RustBufferFromExternal(b RustBufferI) RustBuffer {
```

Comment:
Could you add comments for every exported functions?

The proper Go doc format requires all the exported things (capitalized types or functions) to have a comment. The comment should start with the name of the function/variable.

So for this, it'd be `RustBufferFromExternal (insert descriptive comment)`

Reply:
This folder contains generated code by `uniffi-bindgen-go` so I would advise against adding comments here. My idea was to have `wallet/wallet.go` as the package that the rest of the codebase interfaces with.

Sorry documentation is lackluster as I just wanted it to work for now. I will improve this later this week.

Follow up questions, in terms of project structure:
1. Where would you prefer generated code to be? How about naming conventions?
2. We need somewhere to put static lib files. Should these be included in the repo?

Reply:
Ohh it's the fancy stuff I heard about at the rust meetup.

To answer your questions:
1: If the code is generated, then yeah it makes sense to not touch them. I'd just put a big warning at the start of each file to not touch them and have a readme that states that it's generated code.
2: Would it be ok to compile them from source? If not, I'd just put them in the bdkgo package directory.

Reply:
For 2. it seems that cross-compiling with `cgo` is a big hassle. Maybe we can generate code into an `internal` package? What do you think? Would a readme be sufficient? Otherwise comments will be overwritten everytime we need to re-generate.

Reply:
Closing for now!